### PR TITLE
PretrainedHFTransformer can use GPUs

### LINKED ISSUE #87: `device` arg for `PretrainedHFTransformer` doesn't seem to do anything
### Is there an existing issue for this?

- [X] I have searched the existing issues and found nothing

### Bug description

I first noticed that initializing with `device="cuda"` still ran on my machine's CPUs. I also tried inputting gibberish and that seemed to have no impact (expected errors). After digging in it looked like this argument isn't used.

### How to reproduce the bug

_No response_

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Molfeat version (e.g., 0.1.0):
#- PyTorch Version (e.g., 1.10.0):
#- RDKit version (e.g., 2022.09.5): 
#- scikit-learn version (e.g.,  1.2.1): 
#- OS (e.g., Linux):
#- How you installed Molfeat (`conda`, `pip`, source):
```

</details>

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kkovary**
closes #87 
## Changelogs

-  Fixes #87 where model was not being loaded on GPU for inference.

---

_Checklist:_

- [ ] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._
- [ ] _Add tests to cover the fixed bug(s) or the new introduced feature(s) (if appropriate)._
- [ ] _Update the API documentation if a new function is added, or an existing one is deleted. Eventually consider making a new tutorial for new features._
- [ ] _Write concise and explanatory changelogs below._
- [ ] _If possible, assign one of the following labels to the PR: `feature`, `fix` or `test` (or ask a maintainer to do it for you)._

---

_discussion related to that PR_

### CODE REVIEW

File: molfeat/trans/pretrained/hf_transformers.py [hunk 1]
```diff
@@ -424,7 +430,7 @@ def _embed(self, inputs, **kwargs):
             hidden_state = out_dict["hidden_states"]
             emb_layers = []
             for layer in self.concat_layers:
-                emb = hidden_state[layer].detach().cpu()  # B, S, D
+                emb = hidden_state[layer].detach().to(self.device)  # B, S, D
```
Comment (maclandrol) [inline molfeat/trans/pretrained/hf_transformers.py:433]: I think this should already be on the device.

**Review by maclandrol — APPROVED** [review]
LGTM, thanks !

File: molfeat/trans/pretrained/hf_transformers.py (hunk 1 above)
Comment (maclandrol) [reply molfeat/trans/pretrained/hf_transformers.py:433]: Please confirm, then I would merge.

Comment (kkovary) [reply molfeat/trans/pretrained/hf_transformers.py:433]: I'll have some time to confirm in the next hour. thanks!

Comment (kkovary) [reply molfeat/trans/pretrained/hf_transformers.py]: you're correct, pushed the change
